acceptance: clean up logging

tbg opened the pull request:
Do not create log directories unless asked to.
On TeamCity, put the logs in the artifacts directory (this got lost
during the `pkg` transition). Noticed during triage of #10056.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/10180)

<!-- Reviewable:end -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/acceptance/cluster/localcluster.go, line 183 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180#-KUrzDX_kMVSVAP-q-gM:-KUrzDX_kMVSVAP-q-gN:b4s8nw5) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558c41078f655782772ff9b53b2171a85458203/pkg/acceptance/cluster/localcluster.go#L183)):*
> ```Go
> 	clusterID := uuid.MakeV4()
> 	clusterIDS := clusterID.Short()
> 	var uniqueLogDir string
> ```

this is a bit subtle. specifically, i can't reconstruct the fact that this prevents creation of a directory from reading this code. can you add a comment that explains how and why that happens?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180#-:-KUrzQCjNK2SwrpzRlhx:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/acceptance/cluster/localcluster.go, line 183 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180#-KUrzDX_kMVSVAP-q-gM:-KUs4nGjKv5aJhU-fnkm:b-q6ji97) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558c41078f655782772ff9b53b2171a85458203/pkg/acceptance/cluster/localcluster.go#L183)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is a bit subtle. specifically, i can't reconstruct the fact that this prevents creation of a directory from reading this code. can you add a comment that explains how and why that happens?
</details>
Done (?).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/acceptance/cluster/localcluster.go, line 183 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180#-KUrzDX_kMVSVAP-q-gM:-KUs7NvyYi-NP9NfJzpu:bq44q1x) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558c41078f655782772ff9b53b2171a85458203/pkg/acceptance/cluster/localcluster.go#L183)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> Done (?).
</details>
"LocalCluster will create the log directory if it is nonzero"? IDK, this is probably fine.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: clean up logging

Do not create log directories unless asked.
On TeamCity, put the logs in the artifacts directory (this got lost
during the `pkg` transition). Noticed during triage of #10056.

tschottdorf commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/acceptance/cluster/localcluster.go, line 183 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180#-KUrzDX_kMVSVAP-q-gM:-KUs8EOB8f3AmdqOsJBT:b-5dzj46) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558c41078f655782772ff9b53b2171a85458203/pkg/acceptance/cluster/localcluster.go#L183)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> "LocalCluster will create the log directory if it is nonzero"? IDK, this is probably fine.
</details>
Documented this in the method comment and on the `LocalCluster` struct.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10180)*
<!-- Sent from Reviewable.io -->
